src/realm/array.hpp

```diff
@@ -1050,7 +1049,7 @@ inline void Array::adjust(size_t ndx, int_fast64_t diff)
 {
     REALM_ASSERT_3(ndx, <=, m_size);
     if (diff != 0) {
-        // FIXME: Should be optimized
+        // Should be optimized
```

Comment:
Should these not be "Improvement:" or TODO like you used further up? Similarly with the next one.